### PR TITLE
Added sign up flow

### PR DESCRIPTION
**Pull request by sudheshna-donthineni**

### CODE REVIEW

File: app/id_generation.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            self.get_class_code()
+            + self.get_jnv_code()
+            + self.generate_three_digit_code()
+        )
+
+    def get_id(self):
+        return self.id
+
+    def get_class_code(self):
+        class_codes = {"9": "26", "10": "25", "11": "24", "12": "23"}
```
Comment (suryabulusu) [inline app/id_generation.py:24]: we need to keep a track of this -- keeps on changing year after year :(

File: app/id_generation.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            if len(response.json()) == 0:
+                return HTTPException(
+                    status_code=404, detail="JNV or region does not exist!"
+                )
+            return response.json()
+        return HTTPException(status_code=response.status_code, detail=response.errors)
+
+    def generate_three_digit_code(code=""):
+        for _ in range(3):
+            code += random.randint(0, 9)
```
Comment (suryabulusu) [inline app/id_generation.py:41]: this won't work.. `code` is string. you are adding an int. 
make it `code += str(random.randint(0, 9))

File: app/router/user.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+        if len(response.json()) == 0:
+            return HTTPException(status_code=404, detail="No user found!")
+        return response.json()
+    raise HTTPException(status_code=404, detail="No user found!")
+
+
+def id_generation(data):
+    if data["group"] == "JNVStudents":
+        counter = settings.JNV_counter
+        if counter > 0:
```
Comment (suryabulusu) [inline app/router/user.py:91]: while counter > 0

**Review by suryabulusu — COMMENTED** [review]
left few comments.. looks good

File: app/router/user.py [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+def get_users(request: Request):
+    """
+    This API returns a user or a list of users who match the criteria(s) given in the request.
+
+    Optional Parameters:
+    phone (str), date_of_birth (str), email (str).
+    For extensive list of optional parameters, refer to the DB schema note on Notion.
+
+    Returns:
+    list: user data if user(s) whose details match, otherwise 404
```
Comment (suryabulusu) [inline app/router/user.py:52]: how does a typical return user object look like

File: app/router/user.py [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+            ):
+                raise HTTPException(
+                    status_code=400,
+                    detail="Email/Phone is not part of the request data",
+                )
+            does_user_already_exist = get_users(
+                email=query_params["email"], phone=query_params["phone"]
+            )
+            if not does_user_already_exist:
+                while True:
```
Comment (suryabulusu) [inline app/router/user.py:155]: need to edit this

File: app/router/user.py [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+                        response = requests.post(user_db_url, params=query_params)
+                        if response.status_code == 201:
+                            return query_params["student_id"]
+                        raise HTTPException(status_code=500, detail="User not created!")
+
+            else:
+                response = student.get_students(does_user_already_exist["user_id"])
+                if response.status_code == 200:
+                    return response["student_id"]
+                raise HTTPException(status_code=500, detail="User not created!")
```
Comment (suryabulusu) [inline app/router/user.py:168]: just a doubt -- when exactly will this line occur?

so required user type is student, and id generation is set to true too
then we notice that the given id actually exists. 
so.. shouldn't student.get_students(student_id) return something tangible too?

File: app/id_generation.py (hunk 1 above)
Comment (sudheshna-donthineni) [reply app/id_generation.py:24]: actually, now that I think about it, I can probably automate generate it.

**Pull request merged**